src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -118,32 +119,21 @@ def onExcelCopy(self):
     def onLatexCopy(self):
         self.currentTab.onCopyToClipboard("Latex")
 
-    def serializeAllFitpage(self):
+    def serializeAll(self, all_data):
+        return self.serializeAllFitpage(all_data)
+
+    def serializeAllFitpage(self, all_data):
         # serialize all active fitpages and return
         # a dictionary: {data_id: fitpage_state}
-        params = {}
         for i, tab in enumerate(self.tabs):
-            tab_data = self.getSerializedFitpage(tab)
-            if 'data_id' not in tab_data: continue
-            id = tab_data['data_id'][0]
-            if isinstance(id, list):
-                for i in id:
-                    if i in params:
-                        params[i].append(tab_data)
-                    else:
-                        params[i] = [tab_data]
-            else:
-                if id in params:
-                    params[id].append(tab_data)
-                else:
-                    params[id] = [tab_data]
-        return params
+            all_data = self.getSerializedFitpage(tab, all_data)
+        return all_data
 
-    def serializeCurrentFitpage(self):
+    def serializeCurrentPage(self, all_data):
```

Comment:
See my comment above - let's not modify arguments passed to a method but rather return the modifications.